From 304bb5c43f9f095f01f43ba7f1c842a044be249c Mon Sep 17 00:00:00 2001
From: Zongdong Jiao <zongdong.jiao@amlogic.com>
Date: Fri, 25 Oct 2013 14:33:43 +0800
Subject: [PATCH 4297/5965] PD #81694: hdmitx: audio: fix hdmi cts 7-31 wrong
 AVI issue

Conflicts:

	drivers/amlogic/hdmi/hdmi_tx/hdmi_tx_audio.c
---
 drivers/amlogic/hdmi/hdmi_tx/hdmi_tx_audio.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/amlogic/hdmi/hdmi_tx/hdmi_tx_audio.c b/drivers/amlogic/hdmi/hdmi_tx/hdmi_tx_audio.c
index 918b4652b94d..ed82480ea53d 100755
--- a/drivers/amlogic/hdmi/hdmi_tx/hdmi_tx_audio.c
+++ b/drivers/amlogic/hdmi/hdmi_tx/hdmi_tx_audio.c
@@ -48,7 +48,10 @@ static void hdmi_tx_construct_aud_packet(Hdmi_tx_audio_para_t* audio_param, unsi
         hdmi_print(INF, AUD "Audio Type: PCM\n");
         if(AUD_DB){
 //Note: HDMI Spec V1.4 Page 154
-            AUD_DB[0] = (CT_PCM<<4)|(audio_param->channel_num) ;
+            if((audio_param->channel_num == CC_2CH) || (audio_param->channel_num == CC_REFER_TO_STREAM))
+                AUD_DB[0] = 0;
+            else
+                AUD_DB[0] = (0<<4)|(audio_param->channel_num);
             AUD_DB[1] = (FS_REFER_TO_STREAM<<2)|SS_REFER_TO_STREAM;
             AUD_DB[2] = 0x0;
             if(audio_param->channel_num == CC_6CH)
-- 
2.19.0

